homeassistant/components/aidot/config_flow.py

```diff
@@ -0,0 +1,189 @@
+
+from .const import (
+    CLOUD_SERVERS,
+    CONF_CHOOSE_HOUSE,
+    CONF_PASSWORD,
+    CONF_SERVER_COUNTRY,
+    CONF_USERNAME,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    if len(data["host"]) < 3:
+        raise InvalidHost
+    return {"title": data["host"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle aidot config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.__login_control = LoginControl()
+        self.login_response: dict[Any, Any] = {}
+        self.accessToken = ""
+        self.house_list: list[Any] = []
+        self.device_list: list[Any] = []
+        self.product_list: list[Any] = []
+        self.selected_house: dict[Any, Any] = {}
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                # get ContryCode
+                selected_contry_name = user_input[CONF_SERVER_COUNTRY]
+                selected_contry_obj = {}
+                for item in CLOUD_SERVERS:
+                    if item["name"] == selected_contry_name:
+                        selected_contry_obj = item
+                        break
+                self.__login_control.change_country_code(selected_contry_obj)
+
+                self.login_response = await self.__login_control.async_post_login(
+                    self.hass,
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+                if self.login_response is None:
+                    errors["base"] = "login_failed"
+                    return self.async_show_form(
+                        step_id="user", data_schema=None, errors=errors
+                    )
+                self.accessToken = self.login_response["accessToken"]
+
+                # get houses
+                self.house_list = await self.__login_control.async_get_houses(
+                    self.hass, self.accessToken
+                )
+                if self.house_list is None or len(self.house_list) == 0:
+                    errors["base"] = "get_house_failed"
+                    return self.async_show_form(
+                        step_id="user", data_schema=None, errors=errors
+                    )
+
+                return await self.async_step_choose_house()
+
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidHost:
+                errors["host"] = "cannot_connect"
+            except Exception:
```

Comment:
Is it better to just leave this unhandled; as then at least a backtrace/more context will be retained. Right now it will lose all context.